components/function-controller/internal/webhook/conversion_webhook.go

```diff
@@ -327,8 +339,16 @@ func convertResourcesV1Alpha2ToV1Alpha1(in *serverlessv1alpha2.Function, out *se
 	if in.Spec.ResourceConfiguration != nil && in.Spec.ResourceConfiguration.Build != nil && in.Spec.ResourceConfiguration.Build.Resources != nil {
 		out.Spec.BuildResources = *in.Spec.ResourceConfiguration.Build.Resources
 	}
-	if in.Spec.ResourceConfiguration != nil && in.Spec.ResourceConfiguration.Function != nil && in.Spec.ResourceConfiguration.Function.Resources != nil {
-		out.Spec.Resources = *in.Spec.ResourceConfiguration.Function.Resources
+	if in.Spec.ResourceConfiguration != nil && in.Spec.ResourceConfiguration.Function != nil {
+		if in.Spec.ResourceConfiguration.Function.Resources != nil {
+			out.Spec.Resources = *in.Spec.ResourceConfiguration.Function.Resources
+		}
+		if in.Spec.ResourceConfiguration.Function.Profile != "" {
```

Comment:
We can wrap it in function `convertResourceProfileV1alpha2ToV1Alpha1`

Reply:
changed